src/test/java/org/apache/commons/lang3/reflect/TypeUtilsTest.java

```diff
@@ -543,221 +539,108 @@ public void testIsArrayTypeClasses() {
     @Test
     public void testIsAssignable() throws SecurityException, NoSuchMethodException,
             NoSuchFieldException {
-        List list0 = null;
-        List<Object> list1;
-        List<?> list2;
-        List<? super Object> list3;
-        List<String> list4;
-        List<? extends String> list5;
-        List<? super String> list6;
-        List[] list7 = null;
-        List<Object>[] list8;
-        List<?>[] list9;
-        List<? super Object>[] list10;
-        List<String>[] list11;
-        List<? extends String>[] list12;
-        List<? super String>[] list13;
         final Class<?> clazz = getClass();
         final Method method = clazz.getMethod("dummyMethod", List.class, List.class, List.class,
                 List.class, List.class, List.class, List.class, List[].class, List[].class,
                 List[].class, List[].class, List[].class, List[].class, List[].class);
         final Type[] types = method.getGenericParameterTypes();
-//        list0 = list0;
         delegateBooleanAssertion(types, 0, 0, true);
-        list1 = list0;
```

Comment:
OK, I might see what's going on, in which case these assignments are exactly what's being tested. I'm not sure, but if so they should not be removed

Reply:
Personally, I don't see how assigning a bunch of unused variables here and there could affect the generic types declared on the dummy method/fields in the test. But since removing these local variables doesn't cause any assertion failures, it feels like (A) they're redundant, and/or (B) we have a broken test that doesn't verify what was intended. Should we try to run it on different JVMs (Oracle/Eclise/etc.)?

Reply:
I don't really follow the original code here either. To some extent Chesterton's fence is in play. Since I don't understand why somebody built this fence (or code) I'm much more cautious than usual about removing it. :-)

Reply:
This code was originally written very poorly. It's possible it never tested what i was meant to test, though I am curious why certain lines were commented out as if the test would break if they were uncommented.

The delegateBooleanAssertion method should have been inlined, and the entire method should have been broken up into many test methods with one assert per test.

Possibly doing that now might help us understand it better. It might also help if we could figure out how changing the code here by removing one of the assignments causes the test to break. But maybe it doesn't?

It does seem true that all these assignments you've removed seem like they should have no effect.

Reply:
Aha! I see what these assignments do. They're pseudo-comments, indicating what the next line is testing. They're bad comments, since they're not actually comments and since they're more confusing than not, but they should be safe to remove. 

But wow, is this method bad. The types array is equally confusing. There should be an individual named variable for each type being compared.

If you don't want to go that far, I suggest not deleting these assignments but instead making them comments and maybe expand the comment to something like 

// test the generic type of list 1 is assignable to the generic type of list 2

Reply:
Wow! Thanks for taking the time to crack this puzzle. Using variable assignments as pseudo-comments is a profoundly original way to confuse people 🙂

Here are a few things I did to make this test a little easier to digest:
- Broke it down into 6 test groups. Let me know if you think we should go with smaller groups;
- Inlined `delegateBooleanAssertion` method;
- Removed the unused private `testIsAssignable` method;